src/fu-util.c

```diff
@@ -1593,6 +1593,83 @@ fu_util_report_history_force(FuUtilPrivate *priv, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_report_export(FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	g_autoptr(GPtrArray) devices_filtered =
+	    g_ptr_array_new_with_free_func((GDestroyNotify)g_object_unref);
+	g_autoptr(GPtrArray) devices = NULL;
+
+	/* get all devices from the history database, then filter them and export to JSON */
+	devices = fwupd_client_get_history(priv->client, priv->cancellable, error);
+	if (devices == NULL)
+		return FALSE;
+	g_debug("%u devices with history", devices->len);
+
+	for (guint i = 0; i < devices->len; i++) {
+		FwupdDevice *dev = g_ptr_array_index(devices, i);
+
+		/* filter, if not forcing */
+		if (!fwupd_device_match_flags(dev,
+					      priv->filter_device_include,
+					      priv->filter_device_exclude))
+			continue;
+		if ((priv->flags & FWUPD_INSTALL_FLAG_FORCE) == 0) {
+			if (fwupd_device_has_flag(dev, FWUPD_DEVICE_FLAG_REPORTED)) {
+				g_debug("%s has already been reported", fwupd_device_get_id(dev));
+				continue;
+			}
+			if (!fwupd_device_has_flag(dev, FWUPD_DEVICE_FLAG_SUPPORTED)) {
+				g_debug("%s is not supported", fwupd_device_get_id(dev));
+				continue;
```

Comment:
I would think in a "pure offline" scenario (which this is trying to support) this would always be the case.  Any harm in just having all devices with history?

Reply:
Right, good catch-- we don't want to filter on SUPPORTED at all. FIXED.